### PR TITLE
PostgreSQL support for Docker + Adding entrypoint to isolate build and run phases in Docker

### PR DESCRIPTION
**Pull request by phoenixadb**
### Describe the Pull Request

```
Hi,

We make it works with PostgreSQL support in Docker.
We also added a separate phase for running the container through an entrypoint.
I have update the path for data persistence inside home user directory.
```

### Checklist for PR

- [X] Run MobSF unit tests and lint `tox -e lint,test`.
- [X] Tested Working on Linux, Mac, Windows, and Docker
- [X] Make sure build is passing on your PR. [Build Status](https://travis-ci.com/MobSF/Mobile-Security-Framework-MobSF/pull_requests)

### Additional Comments (if any)

```
Dynamic analysis for Android is still broken. We have not investigated this point as we only use static testings
```

### CODE REVIEW

**Comment by ajinabraham** [comment]
There is conflicting PRs https://github.com/MobSF/Mobile-Security-Framework-MobSF/pull/1380
@superpoussin22 How should we take this forward?

**Comment by superpoussin22** [comment]
@ajin we will use this one I think. Will test quickly ( as soon as my new env is ready)

**Comment by superpoussin22** [comment]
sounds clearly good for me

File: Dockerfile [hunk 1]
```diff
@@ -94,14 +100,14 @@ WORKDIR /root/Mobile-Security-Framework-MobSF
 # Add apktool working path
 RUN mkdir -p /root/.local/share/apktool/framework
 
+ADD scripts .
```
[the remark this answers is not in the document]
Comment (phoenixadb) [reply Dockerfile]: Hi Ajin, I deleted the ADD command and used the script copied earlier

File: Dockerfile [hunk 2]
```diff
@@ -86,6 +86,12 @@ RUN sed -i 's/USE_HOME = False/USE_HOME = True/g' MobSF/settings.py && \
 
 # Postgres support is set to false by default
 ARG POSTGRES=False
+
+ENV POSTGRES_USER=postgres
+ENV POSTGRES_PASSWORD=password
+ENV POSTGRES_DB=mobsf
+#ENV POSTGRES_HOST=postgres
```
Comment (ajinabraham) [inline Dockerfile:93]: Should this be uncommented @phoenixadb ?

File: Dockerfile [hunk 3]
```diff
@@ -86,6 +86,12 @@ RUN sed -i 's/USE_HOME = False/USE_HOME = True/g' MobSF/settings.py && \
 
 # Postgres support is set to false by default
 ARG POSTGRES=False
+
+ENV POSTGRES_USER=postgres
```
Comment (ajinabraham) [inline Dockerfile:90]: should the `ENV` fields be all removed?

Comment (superpoussin22) [reply Dockerfile:90]: For me no, it provide default values if theres no en variable set by user

**Comment by ajinabraham** [comment]
@phoenixadb @superpoussin22 
I followed the instructions here: https://mobsf.github.io/docs/#/docker
```
docker-compose build
docker-compose up -d
```

I am getting the following error on macOS Mojave 10.14.6
```
Starting mobile-security-framework-mobsf_postgres_1 ... error

ERROR: for mobile-security-framework-mobsf_postgres_1  Cannot start service postgres: Mounts denied: { errno = [ENOTDIR]; call = getattrlist; label = /Users/<user>/mobsf/postgresql_data }

ERROR: for postgres  Cannot start service postgres: Mounts denied: { errno = [ENOTDIR]; call = getattrlist; label = /Users/<user>/mobsf/postgresql_data }
ERROR: Encountered errors while bringing up the project.
```

**Comment by superpoussin22** [comment]
think the documentation must be updated. A single docker-compose up -d should work
for the mount can you check if you have /Users under preferences/resources as mount point ?

**Comment by ajinabraham** [comment]
Looks like I had a binary file called `mobsf_data` under `$HOME/MobSF/` causing the issue

**Pull request merged**